src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,30 +35,37 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${selectedTodo?.id}`}
```

Comment:
Here and below - looks that has little sense when `selectedTodo` is `null`

Is you really want to handle this case - probably you should provide some different UI (like text `No Todo is selected`)

But instead I suggest you make `selectedTodo` prop required and change the code accordingly